client/src/app/components/Header/MobileMenu/SideMenu.js

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import "./mobile-menu.css";
+
+class HamburgerMenu extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleTransition = this.handleTransition.bind(this);
+    this.state = {
+      open: false,
+      bar1: undefined,
+      bar2: undefined,
+      bar3: undefined,
+      hide: true
+    };
+  }
+
+  handleTransition() {
+    if (!this.state.open) {
+      this.setState(() => {
+        return {
+          open: true,
+          bar1: "bar-1-animate",
+          bar2: "bar-2-animate",
+          bar3: "bar-3-animate",
+          hide: false
+        };
+      });
+    } else {
+      this.setState(() => {
+        return {
+          open: false,
+          bar1: "bar-1-animate-rev",
+          bar2: "bar-2-animate-rev",
+          bar3: "bar-3-animate-rev",
+          hide: true
+        };
+      });
+    }
+  }
+
+  render() {
+    return (
+      <Nav
+        stateData={this.state}
```

Comment:
'stateData' is not a good name. 

Firstly this name should be agnostic of implementation details such as whether the data comes from state, props, constant etc.

Secondly you should just use a state variable such 'open' and toggle only that. And bar1, bar2, bar3 implementation details will be contained only inside the 'Nav'. You could toggle 'open' class on the parent (#menu), and add the animation classes onPropsChanged (props.open). But you might get away without any div classes, and without any animation classes, and only using CSS selectors like .open div:nth-of-type and use CSS transition instead of animation. Meaning that you won't need to define both ways keyframe animation.This would make the JSX much cleaner